model/mcp/metrics.yaml

```diff
@@ -0,0 +1,68 @@
+groups:
+  - id: mcp.operation.metrics.attributes
+    type: attribute_group
+    brief: MCP request metrics attributes
+    extends: mcp.common.attributes
+    attributes:
+      - ref: mcp.resource.uri
+        requirement_level: opt_in
+
+  - id: mcp.session.metrics.attributes
+    type: attribute_group
+    brief: MCP session metrics attributes
+    attributes:
+      - ref: network.transport
+      - ref: error.type
+        requirement_level:
+          conditionally_required: If and only if session ends with an error.
+
+  - id: metric.mcp.client.operation.duration
+    type: metric
+    metric_name: mcp.client.operation.duration
+    brief: >
+      The duration of the MCP request or notification as observed on the sender
+      from the time it was sent until the response or ack is received.
+    unit: s
+    instrument: histogram
+    stability: development
+    extends: mcp.operation.metrics.attributes
+    attributes:
+      - ref: server.address
+      - ref: server.port
+        requirement_level:
+          recommended: When `server.address` is set
+
+  - id: metric.mcp.server.operation.duration
+    type: metric
+    metric_name: mcp.server.operation.duration
+    brief: >
+      MCP request or notification duration as observed on the receiver
+      from the time it was received until the result or ack is sent.
+    unit: s
+    instrument: histogram
+    stability: development
+    extends: mcp.operation.metrics.attributes
+
+  - id: metric.mcp.client.session.duration
```

Comment:
The number of sessions can be very high. Is this a practical metric?

Reply:
could you elaborate? the number of sessions is not a problem when aggregating session duration.

I have some doubts on the usefulness (it's not super-clear when client should start a new session and whether unrelated operations should happen on the same or different sessions), but assuming session is scoped to one 'logical' operations/conversation the duration is quite useful.

Reply:
Calling it `metric.mcp.client.session.duration` seems like it provides a particular metric for each session. If it's an aggregate, probably `metric.mcp.client.sessions.duration` would be preferable.
I now see that there is no session id or uri involved, therefore the cardinality will likely not explode.

Reply:
this metric follows the same pattern as other metrics (`http.client.request.duration`, `db.client.operation.duration`, `gen_ai.client.operation.duration`, etc) and semconv discourages using pluralization in metric namespaces.